### PR TITLE
feat: Objective-C parser

### PR DESCRIPTION
**Pull request by zippy1978**
- [x] Add Objective-C language
- [x] Generate Objective-C lexer and parser
- [x] Prepare context and visitor
- [x] Create (empty) builtin profile

Dod
- [x] Refactor
- [x] Unit tests
- [x] Documentation (and developer documentation) update

### CODE REVIEW

File: commons/src/test/java/fr/insideapp/sonarqube/apple/commons/issues/JSONRulesDefinitionTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JSONRulesDefinitionTest {
+
+    @Test
+    public void define() {
+
+        JSONRulesDefinition rulesDefinition = new JSONRulesDefinition("rep_key", "rep_name", "lang", "/rules/rules.json") {
+        };
+
```
Comment (gaelfoppolo) [inline commons/src/test/java/fr/insideapp/sonarqube/apple/commons/issues/JSONRulesDefinitionTest.java:33]: Remove one empty line.

**Review by gaelfoppolo — CHANGES_REQUESTED** [review]

File: commons/src/main/java/fr/insideapp/sonaqube/apple/commons/antlr/AntlrContext.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package fr.insideapp.sonaqube.apple.commons.antlr;
```
[the remark this answers is not in the document]
Comment (zippy1978) [reply commons/src/main/java/fr/insideapp/sonaqube/apple/commons/antlr/AntlrContext.java:18]: Nice !
Did not see that :)

File: objc-lang/src/main/java/fr/insideapp/sonarqube/objc/lang/issues/ObjectiveCProfile.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    public static final String OCLINT_RULES_PATH = "/oclint-rules.json";
+
+
+    @Override
+    public void define(Context context) {
+
+        NewBuiltInQualityProfile profile = context.createBuiltInQualityProfile("Objective-C", ObjectiveC.KEY);
+
+        // SwiftLint rules
```
[the remark this answers is not in the document]
Comment (zippy1978) [reply objc-lang/src/main/java/fr/insideapp/sonarqube/objc/lang/issues/ObjectiveCProfile.java:42]: 👍

File: commons/src/test/java/fr/insideapp/sonarqube/apple/commons/issues/JSONRulesDefinitionTest.java (hunk 1 above)
Comment (zippy1978) [reply commons/src/test/java/fr/insideapp/sonarqube/apple/commons/issues/JSONRulesDefinitionTest.java:33]: 👍

File: objc-lang/src/main/java/fr/insideapp/sonarqube/objc/lang/issues/oclint/OCLintRulesDefinition.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import fr.insideapp.sonaqube.apple.commons.issues.JSONRulesDefinition;
+import fr.insideapp.sonarqube.objc.lang.ObjectiveC;
+
+public class OCLintRulesDefinition extends JSONRulesDefinition {
+    public static final String REPOSITORY_KEY = "OCLint";
+    public static final String REPOSITORY_NAME = REPOSITORY_KEY;
+
+    public OCLintRulesDefinition() {
+        super(REPOSITORY_KEY, REPOSITORY_NAME, ObjectiveC.KEY, "/oclint-rules.json");
```
[the remark this answers is not in the document]
Comment (zippy1978) [reply objc-lang/src/main/java/fr/insideapp/sonarqube/objc/lang/issues/oclint/OCLintRulesDefinition.java:28]: Sure !
did it for SwiftLint as well

**Review by gaelfoppolo — APPROVED** [review]

**Pull request merged**